plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-calculator/address.tsx

```diff
@@ -34,16 +37,32 @@ const ShippingCalculatorAddress = ( {
 		( select ) => {
 			return {
 				hasValidationErrors:
-					select( validationStore ).hasValidationErrors,
+					select( validationStore ).hasValidationErrors(),
 				isCustomerDataUpdating:
 					select( CART_STORE_KEY ).isCustomerDataUpdating(),
 			};
 		}
 	);
+	const { defaultFields } = useCheckoutAddress();
+	const formFields = useFormFields(
+		addressFields,
+		defaultFields,
+		'shipping',
+		address.country
+	);
+
+	const hasRequiredFields = () => {
+		const requiredFields = formFields
+			.filter( ( field ) => field.required && ! field.hidden )
+			.map( ( field ) => field.key );
+		return requiredFields.every(
+			( field ) => address[ field ] && address[ field ].trim() !== ''
```

Comment:
I've noted that we need to improve the types of `useFormFields` will maybe raise an issue for that: p1740669343260899-slack-C8X6Q7XQU 

I think though that this code could still be simplified, haven't tested but maybe:


```
const hasRequiredFields = (
  formFields: FormField[],
  address: Address
) =>
  formFields
    .filter((field) => field.required && !field.hidden)
    .every((field) => address[field.key]?.trim() !== "");
```

also doesn't need to be declared inline in the component, how about declaring outside the component and passing `formFields` like I suggested in another comment.

Reply:
I assume that due to bdbae6595ed71e13ab9c88d69f032c8cd9f5e629 this feedback can be marked as resolved, correct?